lua/luasnip/extras/postfix.lua

```diff
@@ -55,6 +55,14 @@ local function generate_opts(match_pattern, user_callback)
 	}
 end
 
+local function wrap_condition(user_condition, trigger, match_pattern)
+	return function(line_to_cursor, ...)
+		return (not user_condition or user_condition(line_to_cursor, ...))
+			and line_to_cursor:sub(1, -1 - #trigger):match(match_pattern)
```

Comment:
Ah, one thing which is a bit annoying:
`context.trig` might also be a pattern, so just removing its length could be an issue.
Also, if `show_condition` is invoked, the trigger is not yet in the file, so it would not have to be removed from `line_to_cursor`.

Creating separate functions for `condition` and `show_condition` should fix both issues, since `condition` provides the matched trigger, and that can then be removed correctly (and show_condition can just check against the entire `line_to_cursor`)

Reply:
I tried to separate them:

```lua
local function wrap_show_condition(user_condition, match_pattern)
	return function(line_to_cursor)
		local value = (not user_condition or user_condition(line_to_cursor))
			and line_to_cursor:match(match_pattern) ~= nil
		print(vim.inspect({line_to_cursor, value}))
		return value
	end
end
```

```lua
	context.show_condition = wrap_show_condition(context.show_condition, match_pattern)
```

But for some reason it's never showing up in the completion menu after this change. Even though I'm seeing the `value` is `true` in the print statement. 🤔

Reply:
I think it's because completion engine will filter out entries that doesn't match with the input. If you typed `count` then trigger completion, `++` won't match with `count`, so it's hidden from completion menu even tho `show_condition` returned `true`.

Reply:
Ohh, that's a bummer :/
Makes sense though.. there should be some ways to customize the fuzzy matching of cmp, maybe we can hijack that somehow.

Reply:
Is it okay to not do anything with `show_condition` right now and just fix `condition`? 🤔

Reply:
Yeah, feel free to :+1:
I think that needs some fix in `cmp_luasnip` anyway

Reply:
Done.